test/common/storagelib.py

```diff
@@ -193,7 +193,7 @@ def dialog_set_val(self, field, val):
 
     def dialog_combobox_choices(self, field):
         return self.browser.call_js_func("""(function (sel) {
-                                               var lis = ph_find(sel).querySelectorAll('li');
+                                               var lis = window.ph_find(sel).querySelectorAll('li');
```

Comment:
`window` is the global object, so do we really need to change all the code like this? Or do you prefer it for clarity?

Reply:
I made a testrun with all changes to testlib.py reverted: https://cockpit-logs.us-east-1.linodeobjects.com/pull-20843-be1fb85d-20240806-133921-arch-storage/log.html

Looks pretty green.

Reply:
Or in other words, this isn't actually a API break.

Reply:
This is a prerequisite for the BiDi conversion in #20832. Exporting globals *only* works with CDP, you can't do it with webdriver or bidi. So that reversion only proves that main isn't broken, but there was never any doubt about that :grin:

Reply:
No, it's not for clarity, it's the core of this change: You *cannot* export globals if all you have is "you define a function and the browser calls it". We have to attach it to an existing global, like sizzle does.

Reply:
Yes, we have to say `window.ph_select = function () { }`, but do we also have to say `window.ph_select()` everywhere? Is `window` not the global object when evaluating JavaScript via bidi?

Reply:
That works! TIL.

Reply:
Yippie!

Reply:
But yeah, testing this without bidi was bogus, urks.